gravitee-am-gateway/gravitee-am-gateway-handler/gravitee-am-gateway-handler-common/src/main/java/io/gravitee/am/gateway/handler/common/email/impl/EmailServiceImpl.java

```diff
@@ -256,11 +259,21 @@ private Map<String, Object> prepareEmailParams(User user, Client client, Integer
             params.put("client", new ClientProperties(client));
         }
 
-        params.put("url", domainService.buildUrl(domain, path, queryParams));
+        params.put("url", domainService.buildUrl(domain, template.redirectUri(), queryParams));
 
         return params;
     }
 
+    private Optional<TokenPurpose> getTokenPurpose(io.gravitee.am.model.Template template) {
+        var purpose = switch (template) {
+            case RESET_PASSWORD -> TokenPurpose.RESET_PASSWORD;
+            case REGISTRATION_VERIFY -> TokenPurpose.REGISTRATION_VERIFY;
+            default -> null;
+        };
+        return Optional.ofNullable(purpose);
+
+    }
```

Comment:
```suggestion
        return switch (template) {
            case RESET_PASSWORD -> Optional.of(TokenPurpose.RESET_PASSWORD);
            case REGISTRATION_VERIFY -> Optional.of(TokenPurpose.REGISTRATION_VERIFY);
            default -> Optional.empty();
        };
    }
```
Also is there a difference in the logic between `TokenPurpose.UNSPECIFIED` and `null` ?

Reply:
`EmailService` uses `null` because if we don't know what the purpose is it doesn't make sense to put the claim in the token.
On the other hand, `RedirectUriValidationHandler` logic gets simpler if it can assume that every token has a non-null purpose - that's why `TokenPurpose.of(String)` returns `UNSPECIFIED` instead of null/`Optional.empty()`